### ISSUE TITLE
Feature: Prototype UI for Hourly Apportioned Emissions Required Fields Filter Criteria 

### ISSUE THREAD

**Issue by JorjaComer** [US-EPA-CAMD/PI-1#172]
## Feature Benefit
Determine data publishing solution, both from a technical solution (how data is available) and business perspective (what data is available)

The following are notes for discussion. The functionality described in the user story is always the source of truth.

### Hourly Emissions Filter/Sort 
**_Questions:_**

1. Will the site use the EPA header/footer? Will it use the go wide template? - Waiver process has begun. Header and content are fairly defined. Footer is TBD. Development will initially focus on main content, filter sort criteria, as template work continues in parallel. Pages should be responsive to screen size. 
2. What happens to already entered filter criteria when the data type or sub data type is changed?  - clear criteria 

**_Assumptions:_**

1. Blank parameter values will return all records 
2. Query will be executed as an AND search (i.e. return only records that match all criteria)
3. Download limits will be implemented in future and are out of scope for initial development 
4. Users must select data type and data subtype before other fields are available 
5.The site will be responsive to different screen sizes 

**_Parameters_**
**Data Type**
Button/drop down
Value: Emissions, Allowance, Compliance, default blank, 
Required
Single select 

**Data Sub-Type**
Drop down 
Value: Hourly Emissions, Daily Emissions, Monthly Emissions, Quarterly Emissions, Ozone Season Emissions, Annual Emissions, 
Facility/Unit Attributes, default blank
Required
Single select 

**Time Period**
Date picker (USWDS date picker)
Format: MM/DD/YYYY
Required
Valid date range: 01/01/1995 to current quarter
Only one date range may be selected at a time (annual data from 1980, not included in hourly query)

**_Program_**	
Drop down, default blank
Values: All active and inactive programs (MATS is excluded)	
Optional
Multi select 
Out of scope: If facility is already selected, then programs is limited to programs associated to the selected facility 	

**Facility**
**Facility Name**
Dynamic search
Search is a contains search
Results are displayed in Facility Name (ORIS Code)
Results populate in a list, parameters populates when a value is selected from the list  	(results list should stay at top of page/visible in current view, possible use pagination to keep window small)
Search is not case sensitive
Out of scope: If state is already selected, then search is limited to facilities in the selected state 	
Out of scope: If program is already selected, then facilities is limited to facilities associated to the selected program	
If no results are found, the system will indicate so. “No facilities match your search.”
Multi select 

**ORIS Code**	
Dynamic search
Search is a contains search
Results populate in a list, parameters populates when a value is selected from the list 
Results are displayed in Facility Name (ORIS Code) format
Out of scope: If state is already selected, then search is limited to facilities in the selected state 	
Out of scope: If program is already selected, then facilities is limited to facilities associated to the selected program
If no results are found, the system will indicate so. “No facilities match your search.”
Multi Select 

**State**
Dynamic search
Search is a “begins with” search
Results populate in a list, parameters populates when a value is selected from the list  	
Results display the full state name (Texas, not TX)
Out of scope: If Facility/ORIS Code is already selected, then state is automatically populated with the state (or states) associated to the facility(or facilities) 
Multi Select

**Unit Characteristics** 

**Unit Type**
Drop down
Values: Boiler, Turbine, default blank 
Optional
Multi select

- If boiler is selected, then the system displays a list of boiler unit types, multi select
- If a user selects All Boilers, then the system will show all boiler types are selected. 
- The system will provide the ability to select and deselect boiler types. 
- If turbine is selected, then the system displays a list of turbine unit types, multi select
- If a user selects All Turbines, than the system will show all turbine types are selected. 
- The system will provide the ability to select and deselect turbine types. - 

**Primary and Secondary Unit fuel Types** **Note: need to add sub fuel type selections. Refer to current AMPD for list**
Drop down
Values: Coal, Gas, Oil, Other, default blank 
Optional
Multi select
Search is performed on Primary and Secondary fuel types  NOTE: Add "?" text to inform user this is how search works 
If coal is selected, then the system displays a list of coal fuel types, multi select
If gas is selected, then the system displays a list of gas fuel types, multi select
If oil is selected, then the system displays a list of oil fuel types, multi select

**Control Technologies**
Drop down
Values: SO2 Controls, Hg Controls, NOx Controls, PM Controls, default blank 
Optional
Multi Select 
If SO2 is selected, then the system displays a list of SO2 control technologies, multi select
If Hg is selected, then the system displays a list of Hg control technologies, multi select
If NOx is selected, then the system displays a list of NOx control technologies, multi select
If PM is selected, then the system displays a list of PM control technologies, multi select

**_Change Criteria_**
The user will have the ability to clear one, many, or all filter criteria
If the user changes data type or sub data type, then:

- The filter criteria displayed updates accordingly 
- No previously entered filter criteria is retained 

If a user changes data type or sub data type, then filter criteria is cleared 
If the user chooses to clear one or more filter criteria before preview is selected, then the system clears the criteria from the filter and updates any remaining dependent criteria accordingly.
If the user chooses to clear one or more filter criteria after preview is selected, then the system clears the criteria from the filter AND download is disabled until data is previewed. 

**_Data Preview_**
Trigger: A user has entered, at a minimum, the required parameters, and selected the option to preview data.
For each record that matches the criteria, the system will display in a tabular format in the following order:

STATE| FACILITY_NAME| ORISPL_CODE| UNITID | ASSOC_STACKS | OP_DATE |OP_HOUR| OP_TIME | GLOAD | SLOAD | SO2_MASS | SO2_MASS_MEASURE_FLG | SO2_RATE | SO2_RATE_MEASURE_FLG | NOX_MASS | NOX_MASS_MEASURE_FLG | NOX_RATE | NOX_RATE_MEASURE_FLG | CO2_MASS |  CO2_MASS_MEASURE_FLG | CO2_RATE | CO2_RATE_MEASURE_FLG | HEAT_INPUT| PRIMARY_FUEL_INFO|
SECONDARY_FUEL_INFO | UNIT_TYPE_INFO| SO2_CONTROL_INFO | PART_CONTROL_INFO | NOX_CONTROL_INFO | HG_CONTROL_INFO |
PRG_CODE_INFO

Refer to the latest CAMPD_DataLabels_Hourly Emissions [here](https://teams.microsoft.com/_#/files/General?threadId=19%3A6f05dfa026c84672a2cc1bfa72c58f14%40thread.skype&ctx=channel&context=General%2520Overview&rootfolder=%252Fsites%252FCAMDCVPTeam%252FShared%2520Documents%252FGeneral%252FAnalysis%252C%2520User%2520Research%2520and%2520Design%252FAnalysis%252FGeneral%2520Overview) for column names/headers for the results table 

Results will display record count 
Results are limited to 100 records 
Results in preview will have a scroll functionality to show all columns/rows 
NOTE: NEED TO ID COLUMNS FOR HOVER TEXT AND DISPLAY TEXT 
OUT OF SCOPE: Results may be sorted in alphabetical ascending and descending order
CAN THEY BE FILTERED? IF SO BY WHAT CRITERIA? - no additional filtering 
OUT OF SCOPE: Results are paginated

**_File Selection_**
Trigger: The user has at least one record that matches the search criteria, and has previewed the results
The user has the option to select the file output in JSON or CSV format. CSV is the default file type 
Required 

**_Download_**
Trigger: The user has selected a file output type
When the user selects the option to download, the system creates a file in the format specified  containing all records that match the parameters and downloads the file 
After a user has chosen to download data, then the system will display the preview of the downloaded results.

**Comment by JorjaComer** [comment]
See notes

https://usepa.sharepoint.com/sites/CAMDCVPTeam/_layouts/15/Doc.aspx?sourcedoc={ebc3fb4d-ba1e-45cb-ac7c-095db55470fd}&action=edit&wd=target%28Requirements.one%7C1d91d228-1ada-4717-a664-cb65edf297c9%2F12%5C%2F07%5C%2F2020%20PI%20Planning%20Part%20II%7C3219d9ff-0479-4448-adb6-dfb6222abd97%2F%29&wdorigin=703

**Comment by JorjaComer** [comment]
Split feature. See #1100 for additional work needed to complete the filter sort for hourly emissions end to end functionality

**Comment by JorjaComer** [comment]
Moved all stories, bugs, and tasks for work in PI I&P (3.6) and PI 4 to #1100 . All PI 3 issues are complete.
